### DISCUSSION TITLE
Influxdb instrumentation

### DISCUSSION THREAD

**Discussion by abhisesh** [open-telemetry/opentelemetry-java-instrumentation#8072] [Q&A] [answered]
Hi team,

My question relates to #1572. Wanted to submit a PR for this, but while digging through the influxdb java library code, I noticed that `okhttp` and `retrofit` libraries are used internally for sending http requests. `retrofit` is also using `okhttp` internally.

Here's an example: https://github.com/influxdata/influxdb-client-java/blob/master/client/src/main/java/com/influxdb/client/internal/AbstractInfluxDBClient.java#L75

So does it make sense to implement custom instrumentation for Influxdb, since okhttp instrumentation is already supported?

**Comment by laurit** [comment, marked as answer]
OkHttp instrumentation provides telemetry with [http semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/http/) for influxdb client [database semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md) might be more appropriate. If you believe that database semantic conventions would provide better telemetry then implementing dedicated instrumentation for influxdb client is the way to go.
